terraform-dev/main-gcp.tf

```diff
@@ -182,6 +186,17 @@ resource "google_compute_project_default_network_tier" "default" {
 resource "google_project_iam_policy" "project" {
   project     = var.project_id
   policy_data = data.google_iam_policy.project.policy_data
+
+
+  # Add the necessary role and member for Redis service account
+  bindings = [
+    {
+      role    = "roles/redis.serviceAgent"
+      members = [
+        "serviceAccount:service-${var.project_number}@cloud-redis.iam.gserviceaccount.com"
+      ]
+    },
+  ]
```

Comment:
Confusingly, the bindings can't be defined in `google_project_iam_policy`, they have to be defined in `google_iam_policy.project`, which is below.  The syntax is also slightly different there.  I would put the following at about line 370.

```tf
  # Add the necessary role and member for Redis service account
  binding {
    role = "roles/redis.serviceAgent"
    members = [
      "serviceAccount:service-${var.project_number}@cloud-redis.iam.gserviceaccount.com"
    ]
  }
```